Add is_transaction_active() to all connectors

MeanSquaredError opened the pull request:
Currently the MySQL connector provides a handy method called `is_transaction_active()` which allows the user to check if the database connection is running a transaction, without making a round trip to the server and without using database-specific commands or API calls. The PostgreSQL and SQLite3 connectors keep internally information about the transaction state, but do not allow the user to query the transaction state.

This PR unifies the connector API by adding a public method `is_transaction_active()` to all the connectors.

The internal state of the SQLite3 connector supports 3 values, `tx active`, `tx not active` and `maybe`, where `maybe` is the state set while the transaction is being started/stopped. However these connector methods are not re-entrant or multithread-safe and the SQLite3 state variable is not atomic, so it does not make sense to keep a transient "maybe" state which is used only inside the transaction methods without being exported outside to the other methods or to the user. That is why this PR replaces the SQLite3 transaction state with a boolean variable, just like the one used for MySQL and PostgreSQL.

The PR also adds tests for `is_transaction_active()`

The PR was built and tested with
```
cmake -B build -DBUILD_POSTGRESQL_CONNECTOR=ON -DBUILD_SQLITE3_CONNECTOR=ON -DBUILD_MYSQL_CONNECTOR=ON -DBUILD_TESTING=ON -DUSE_SYSTEM_DATE=ON -DDEPENDENCY_CHECK=ON
cmake --build build
cd build
ctest
```
All tests passed successfully.

rbock reviewed (changes requested):
Looks like you are modeling the strategy for setting `_transaction_active = false` after the mysql connector.
I wonder if that is correct? The transaction is certainly still active before sending `COMMIT`?

MeanSquaredError commented:
OK, I changed the transaction shutdown sequence to:
1. Report transaction shutdown. This is done for rollback only.
2. Execute SQL statement that shuts down the transaction.
3. Sets the _transaction_active flag to false.

Originally I used the MySQL order in all connectors just to have consistent order in all connectors. My rationale for using the MySQL order was that it didn't really matter since the connector methods were not re-entrant, so what really mattered was the value of `_transaction_active` before and after the method call.

But after your comment I though that you are right, the MySQL order is wrong, because an exception could occur while executing the SQL statement that stops the transaction and then connection object would be left in invalid state, the transaction would be active, but `_transaction_active` would be false. So I changed everything to execute the SQL statement before clearing the transaction flag.

rbock commented:
Thanks for the update! Looks good to me.
